testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -4568,6 +4568,25 @@ private static void checkIf(SqlOperatorFixture f) {
     f.checkNull("regexp_contains(cast(null as varchar), cast(null as varchar))");
   }
 
+  @Test void testRegexpExtractFunc() {
+    final SqlOperatorFixture f =
+        fixture().setFor(SqlLibraryOperators.REGEXP_EXTRACT).withLibrary(SqlLibrary.BIG_QUERY);
+
+    f.checkString("regexp_extract('abc def ghi', 'def')", "def", "VARCHAR NOT NULL");
+    f.checkString("regexp_extract('abcadcaecghi', 'a.c', 1, 3)", "aec", "VARCHAR NOT NULL");
+    f.checkString("regexp_extract('abcadcaecghi', 'abc(a.c)')", "adc", "VARCHAR NOT NULL");
+    f.checkString("regexp_extract('55as56664as422', '\\d{3}')", "566", "VARCHAR NOT NULL");
+
+    f.checkNull("regexp_extract('abc def ghi', 'asd')");
+    f.checkNull("regexp_extract('abc def ghi', cast(null as varchar))");
```

Comment:
Can you add more tests that check the 3rd and 4th args (if they are out of range, null, etc)

Reply:
Added those as unit tests in the SqlFunctionsTest and big-query.iq file, i didnt add many here as they seemed redundant at every location should I include some here as well?

Reply:
Good question. I am a bit biased because I write most of my tests in `SqlOperatorTest` when adding a new operator. Might be best to get a second opinion.

Reply:
I think it might be best to add some to `SqlOperatorTest` because that seems to serve as the main source of truth for how operators handle their edge cases, I don't necessarily think it's redundant. The operator tests check and showcase that the operator follows the behavior it should, while the function tests show that the actual implementation abides by that behavior as well. What do you think?

Reply:
Sure Tanner, that makes sense we have test cases at each wrapper level so its best to validate at every stage, will add some here.

Reply:
Added additional cases to cover the optional arguments, appreciate the feedback!